src/rgw/rgw_file.h

```diff
@@ -1171,9 +1173,21 @@ namespace rgw {
       return fhr;
     } /*  lookup_fh(RGWFileHandle*, const char *, const uint32_t) */
 
-    inline void unref(RGWFileHandle* fh) {
+    inline void unref(RGWFileHandle* fh, uint32_t flags = RGWFileHandle::FLAG_LOCK) {
```

Comment:
I don't think flags = RGWFileHandle::FLAG_LOCK can be the unmarked (default) case of unlock;  it simply should not be taking these locks by default, I don't think.  This needs to happen in a call path specific to removing, not in unref().

Reply:
Unref() has lru object remove operation, and this need lru lock protecting by default. If that is the last unref (rgw_umount), unref() is safe. Specific path for removing in a callbcak, you mean, splitting `todo->delete` from lru::unref to a specific path?